spec/models/account_spec.rb

```diff
@@ -1,34 +1,87 @@
 describe Account do
-  before(:each) do
-    @user = FactoryGirl.create(:account_user)
-    @group = FactoryGirl.create(:account_group)
+  let(:user) { FactoryGirl.create(:account_user) }
+  let(:group) { FactoryGirl.create(:account_group) }
+
+  context 'when host has users and groups' do
+    let(:host) { FactoryGirl.create(:host) }
+    let(:test_xml) do
+      REXML::Document.new("<miq>
+                          <accounts>
+                          <users><user name='test_user'/></users>
+                          <groups><group name='test_group'/></groups>
+                          </accounts>
+                          </miq>")
+    end
+
+    it '.add_elements' do
+      user.class.add_elements(host, test_xml)
+      expect(host.users.find_by(:name => 'test_user')).to be
+      expect(host.groups.find_by(:name => 'test_group')).to be
+    end
   end
 
   it '#accttype_opposite' do
-    expect(@user.class.accttype_opposite('user')).to eq('group')
-    expect(@group.accttype_opposite).to eq('user')
+    expect(group.accttype_opposite).to eq('user')
+    expect(user.accttype_opposite).to eq('group')
   end
 
-  it '#users raise an error when called on user model' do
-    expect { @user.users }.to raise_error(RuntimeError, "Cannot call method 'users' on an Account of type 'user'")
+  it 'users raise an error when called on user model' do
+    expect { user.users }.to raise_error(RuntimeError, "Cannot call method 'users' on an Account of type 'user'")
   end
 
-  it '#users return empty array when called on group' do
-    expect(@group.users).to be_empty
+  it 'users return empty array when called on group' do
+    expect(group.users).to be_empty
```

Comment:
For this and all tests that look similar, prefer separating method under test from context from expectation:

```ruby
describe "#users" do
  context "when called on a user account" do
    it "raises an error" do
      # ...
    end
   
  context "when called on a group account" do
    it "returns an empty array"
      # ...
    end
  end
end

# etc, etc...
```

The main goal is to group like-contexts or method tests. I prefer the previous example, but alternatively you could group by condition (the type of group):

```ruby
context "with a user account" do
  describe "#users" do
    it "raises an error"
      # ...
    end
  end
  
  describe "#groups" do
    it "returns an empty array" do
      # ...
    end
  end
end

# etc, etc...
```